OpenRA.Mods.Cnc/Traits/Spreads.cs

```diff
@@ -0,0 +1,73 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Any actor with this trait enabled has a chance to affect others with this trait.")]
+	public class SpreadsInfo : ConditionalTraitInfo
+	{
+		[Desc("The chance this actor is going to affect an adjacent actor.")]
+		public readonly int Probability = 10;
+
+		[Desc("How far the condition can spread from one actor to another.")]
+		public readonly WDist Range = WDist.FromCells(3);
+
+		[GrantedConditionReference]
+		[Desc("Condition to grant onto another actor in range with the same trait.")]
+		public readonly string SpreadCondition = "spreading";
+
+		[Desc("Time in ticks to wait between spreading further.")]
+		public int Delay = 5;
+
+		public override object Create(ActorInitializer init) { return new Spreads(this); }
+	}
+
+	public class Spreads : ConditionalTrait<SpreadsInfo>, ITick
```

Comment:
Thinking about it, calling this `SpreadsCondition` might be more intuitive. (`Spreads` what?)

Reply:
I also moved the trait to `Mods.Common` as it contains no special C&C logic.